sound/soc/sof/intel/hda-loader.c

```diff
@@ -101,15 +138,7 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 		goto err;
 	}
 
-	/* DSP is powered up, set all SSPs to slave mode */
-	for (i = 0; i < chip->ssp_count; i++) {
-		snd_sof_dsp_update_bits_unlocked(sdev, HDA_DSP_BAR,
-						 chip->ssp_base_offset
-						 + i * SSP_DEV_MEM_SIZE
-						 + SSP_SSC1_OFFSET,
-						 SSP_SET_SLAVE,
-						 SSP_SET_SLAVE);
-	}
+	hda_ssp_resume(sdev);
```

Comment:
the commit message mentions a different name so that needs to be modified.

what was wrong with  hda_set_ssps_slave()?

Reply:
not inclusive

Reply:
this calls SSP_SET_SLAVE...

I am all for inclusive terms, but this has nothing to do with resume operations.

Reply:
> this calls SSP_SET_SLAVE...
> 
> I am all for inclusive terms, but this has nothing to do with resume operations.

I think Ranjani means "what we need to perform to SSPs when resuming audio device (D3->D0)", like "_ssp_at_resume()". And, use the more generic _ssp_resume() will be friendly to add more SSP related reset/resume/init stuff into it in future when needed.

I am fine on both, just let me know the aligned one.

Reply:
just because we do set_slave doesnt mean we have to name the function that too no? Someday maybe set_slave will get renamed and we wont have to worry about the static function name having the word"slave" in it. But I agree resume is a bad name. This is the loader, so while it is also called during resume, thats not the only thing. Why not call is something lika hda_ssp_cl_init() and hda_sdw_cl_init()

Reply:
the acronym we came up with is CBP Codec BitClockProvider, so we can call this wrapper

hda_ssp_set_cbp()

if it also sets the fsync hda_ssp_set_cbp_cfp().

and no wrapper for SoundWire (see comment below) would be my preference.

Reply:
sound/soc/sof/intel/hda-loader.c:                                                SSP_SET_SLAVE,
sound/soc/sof/intel/hda-loader.c:                                                SSP_SET_SLAVE);
sound/soc/sof/intel/hda.h:#define SSP_SET_SLAVE         (SSP_SET_SCLK_SLAVE | SSP_SET_SFRM_SLAVE)

This should be SSP_SET_CBP_CFP